admin/user.go

```diff
@@ -82,6 +82,25 @@ func (s *Service) CreateOrUpdateUser(ctx context.Context, email, name, photoURL
 		if err != nil {
 			return nil, err
 		}
+
+		for _, usergroupID := range invite.UsergroupIDs {
+			// check if the user group exists, need to check explicitly as tx is not completed yet
+			exists, err := s.DB.CheckUsergroupExists(ctx, usergroupID)
+			if err != nil {
+				return nil, err
+			}
+
+			if !exists {
+				// ignore if usergroup does not exist, might have been deleted before invite was accepted
+				continue
+			}
+
+			err = s.DB.InsertUsergroupMemberUser(ctx, usergroupID, user.ID)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
Maybe for safety, gracefully handle if the user is already a member (I guess it would be a `database.ErrNotUnique`)? In case of duplicates in the list or retried calls

Reply:
this is happening inside a transaction so in case of err tx fails, this needs to be checked explicitly. This is first time login case so If we deduplicate during adding the groups to invite, I think it should not be required to check ?

Reply:
since this is happening inside transaction then in case of retries if usergroup was updated then invite would have been deleted as well so should not be a problem

Reply:
Yeah I guess it's fine then :)